testsuite/tests/libratest/smoke_test.rs

```diff
@@ -746,3 +756,60 @@ fn test_e2e_reconfiguration() {
         Decimal::from_str(&client_proxy_0.get_balance(&["b", "0"]).unwrap()).ok()
     );
 }
+
+#[test]
+fn test_client_waypoints() {
+    let (mut env, mut client_proxy) = setup_swarm_and_client_proxy(3, 1);
+    // Make sure some txns are committed
+    client_proxy.create_next_account(false).unwrap();
+    client_proxy
+        .mint_coins(&["mintb", "0", "10"], true)
+        .unwrap();
+
+    // Create the waypoint for the initial epoch
+    let genesis_li = client_proxy
+        .latest_epoch_change_li()
+        .expect("Failed to retrieve genesis LedgerInfo");
+    assert_eq!(genesis_li.ledger_info().epoch(), 0);
+    let genesis_waypoint = Waypoint::new(genesis_li.ledger_info())
+        .expect("Failed to generate waypoint from genesis LI");
+
+    // Start another client with the genesis waypoint and make sure it successfully connects
+    let mut client_with_waypoint = env.get_validator_ac_client(0, Some(genesis_waypoint.clone()));
+    assert_eq!(
+        client_with_waypoint.latest_epoch_change_li().unwrap(),
```

Comment:
this feels fragile, I looked at how this works and realize it depends on the client_proxy loads the local faucet account which triggers a `get_account_data_from_address` request. although the client is not likely to be created with remote faucet in smoke test.